sdk/core/loader/boot.cc

```diff
@@ -460,11 +460,44 @@ namespace
 					    (size == (LA_ABS(__export_mem_heap_end) -
 					              LA_ABS(__export_mem_heap))))
 					{
+						Debug::log("Rounding heap ({}--{}) region",
+						           target,
+						           target + size);
+						// FIXME: This should not be hard coded.
+						static constexpr size_t MantissaBits = 9;
+						static constexpr size_t MantissaMask =
+						  (1 << MantissaBits) - 1;
+						size_t sizeBits = 32 - __builtin_clz(size);
+						size_t sizeMask = (sizeBits <= MantissaBits)
+						                    ? MantissaMask
+						                    : MantissaMask
+						                        << (sizeBits - MantissaBits);
+						Debug::log("Applying mask {} to size", sizeMask);
+						size_t roundedSize = size & sizeMask;
```

Comment:
Isn't that equivalent to
```suggestion
						size_t sizeMask =
							__builtin_cheri_representable_alignment_mask(size);
						Debug::log("Applying mask {} to size", sizeMask);
						size_t roundedSize = size & sizeMask;
```

The intermediate steps are different, in that CRAM is going to give you all ones to the left of your `MantissaMask`. In the end, though, that shouldn't matter, because all those bits in `size` are zero from the `__builtin_clz()`, and so they'll be zero in `roundedSize`.

Reply:
Yup, that works nicely.